### PR TITLE
chore: merge ccxt upstream to our fork

### PR DESCRIPTION
**Pull request by NeOMakinG**
I did tests every exchanges we have in production, here is the testing table:

https://www.notion.so/cede-labs/Manual-testing-reports-a222148adf0042aeaa504050e7b673b4?p=caaf97019c534c28ac199a525a8b724a&pm=s

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,246 +1,249 @@
...
+  "readme": "README.md",
+  "scripts": {
+    "docker": "docker-compose run --rm ccxt",
+    "fixTSBug": "node build/fixTSBug",
+    "build-docs": "node jsdoc2md.js && node examples2md.js",
+    "serve-docs": "docsify serve ./wiki",
+    "tsBuild": "tsc || true",
+    "tsBuildExamples": "tsc -p ./examples/tsconfig.json",
+    "emitAPI": "node build/generateImplicitAPI.js",
+    "build": "npm run pre-transpile && npm run transpile && npm run post-transpile && npm run update-badges && npm run build-docs",
```
Comment (neeeekitos) [inline package.json:32]: We don't need `post-transpile` script on the build, since we don't use php/py

File: ts/src/coinbase.ts [hunk 2]
```diff
@@ -3133,6 +3238,101 @@ export default class coinbase extends Exchange {
...
+         * @description make a withdrawal
+         * @see https://docs.cloud.coinbase.com/sign-in-with-coinbase/docs/api-transactions#send-money
+         * @param {string} code unified currency code
+         * @param {float} amount the amount to withdraw
+         * @param {string} address the address to withdraw to
+         * @param {string} [tag] an optional tag for the withdrawal
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @returns {object} a [transaction structure]{@link https://docs.ccxt.com/#/?id=transaction-structure}
+         */
+        [ tag, params ] = this.handleWithdrawTagAndParams (tag, params);
```
Comment (neeeekitos) [inline ts/src/coinbase.ts:3254]: Finally, ccxt implemented this method! @NeOMakinG we could create a ticket to migrate to this method on our side.

File: ts/src/bybit.ts [hunk 3]
```diff
@@ -3,17 +3,17 @@
...
+import type { Int, OrderSide, OrderType, Trade, Order, OHLCV, FundingRateHistory, OpenInterest, OrderRequest, Balances, Str, Transaction, Ticker, OrderBook, Tickers, Greeks, Strings, Market, Currency, MarketInterface } from './base/types.js';
 
 //  ---------------------------------------------------------------------------
 
 /**
  * @class bybit
- * @extends Exchange
+ * @augments Exchange
  */
 export default class bybit extends Exchange {
```
Comment (neeeekitos) [inline ts/src/bybit.ts:18]: @nicecastle I don't think this update will fix bybit accounts issue. I didn't find any related changes.

File: ts/src/gate.ts [hunk 4]
```diff
@@ -1929,10 +1927,13 @@ export default class gate extends Exchange {
          * @see https://www.gate.io/docs/developers/apiv4/en/#generate-currency-deposit-address
          * @param {string} code unified currency code
          * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @param {string} [params.network] unified network code (not used directly by gate.io but used by ccxt to filter the response)
          * @returns {object} an [address structure]{@link https://docs.ccxt.com/#/?id=address-structure}
          */
         await this.loadMarkets ();
         const currency = this.currency (code);
+        const rawNetwork = this.safeStringUpper (params, 'network');
+        params = this.omit (params, 'network');
```
Comment (neeeekitos) [inline ts/src/gate.ts:1936]: @NeOMakinG I think this can break our network mappings. Did you check this part?

File: ts/src/gate.ts [hunk 5]
```diff
@@ -1954,34 +1955,39 @@ export default class gate extends Exchange {
...
-            if (chain === network) {
-                networkAddress = multichainAddress;
-                break;
-            }
-        }
-        if (networkAddress === undefined) {
-            throw new BadRequest (this.id + ' fetchDepositAddress() could not find address for network ' + network);
-        }
-        const addressField = this.safeString (networkAddress, 'address');
+        const networkId = this.networkCodeToId (rawNetwork, code);
```
Comment (neeeekitos) [inline ts/src/gate.ts:1958]: Same here

File: ts/src/okx.ts [hunk 6]
```diff
@@ -874,8 +880,8 @@ export default class okx extends Exchange {
                     'ALGO': 'Algorand',
                     'BHP': 'BHP',
                     'APT': 'Aptos',
-                    'ARBONE': 'Arbitrum one',
-                    'AVAXC': 'Avalanche C-Chain',
+                    'ARBONE': 'Arbitrum One',
+                    'AVAXC': 'Avalanche C',
```
Comment (neeeekitos) [inline ts/src/okx.ts:884]: @NeOMakinG I think we should update our mappings for networks here. Could you please check it?

File: ts/src/okx.ts [hunk 7]
```diff
@@ -4431,12 +4457,13 @@ export default class okx extends Exchange {
         const networksById = this.indexBy (networks, 'id');
         const networkData = this.safeValue (networksById, chain);
         const network = this.safeString (networkData, 'network');
+        const networkCode = this.networkIdToCode (network, code);
```
Comment (neeeekitos) [inline ts/src/okx.ts:4460]: We should check this part as well. Our mappings are used to map OKX's raw network values or CCXT's values?

File: ts/src/okx.ts (hunk 6 above)
Comment (NeOMakinG) [reply ts/src/okx.ts:884]: @neeeekitos We don't use this internal mapping for OKX, we already have `Arbitrum One` inside our mapping, it seems like it works as it should

File: ts/src/okx.ts (hunk 7 above)
Comment (NeOMakinG) [reply ts/src/okx.ts:4460]: Fixed

**Review by nicecastle — APPROVED** [review]
all good

**Pull request merged**